docs/DeveloperGuide.md

```diff
@@ -6,8 +6,248 @@
+
+Step 2: The `Command Parser` Class would run the `parse` method and create an `AddItemCommand` object containing an `Item` object which is made up of the name of the item to add as well as its pax which are found within the user input.
+
+Step 3: The `AddItemCommand` class would be passed back to the `Duke` class.
+
+Step 4: The `execute` method of the `AddItemCommand` class would be run and the `ItemList` Object would be extracted from the `List Container` object and the `AddItemCommand` object would pass its Item object to the `ItemList` Object's `addItemToList` method.
+
+Step 5: The `addItemToList` method would then call the `checkForItemDuplicates` method to check if there are any items within the item list with the same name as the item we are about to add.
+
+Step 6:I f there are is an item with a matching name found, we would then call the `UI` object and execute the `UI`'s `printItemAlreadyInTheListErrorMessage` method which would print a message informing the user that the item the user wants to add is already found within the item list and hence nothing would be added. Step 7 and 8 are skipped.
+
+Step 7: If there are no items with a matching name found, we would add the `Item` object to its ArrayList of `Item` objects called listOfItems. The `AddItemCommand` then call the `UI` and execute the `UI`'s `printAddItemAcknowledgementMessage` method which would print an acknowledgement message to the user informing him that the item has been added into the item list.
+
+Step 8: The `ItemListFileManager` object would be called and we would pass the item list to its `writeItemListToFile` method to write the updated item list into the file ListFolder/ItemList.txt.
+
+## Implementation
+
+#### Assign a housekeeper to a room
+The assignment of housekeeper to a room is facilitated by `AssignHousekeeperCommand`. This command objects uses `AssignmentMap`, which keeps a `HashMap`, to help us map a `Room` object to a `Housekeeper` object. 
+
+``
+Map<Room, Housekeeper> map = new HashMap<>();
+``
+
+The following steps exhibit how `AssignHousekeeperCommand` 
+and `AssignmentMap` fit together in the structure and how the mechanism for adding an assinment behaves at each step.
+
+__Step 1.__ The user launches the application. In the `Duke` class, an empty instance of the `AssignmentMap` class,
+called `assignmentMap`, is created.
+
+![Step 1](team/aiman_assignment/step1.png)
+
+__Step 2.__ The user types the command `Assign Susan ## 301`. In the `Duke` class, a `Command` object
+is created by invoking the `CommandParser` class's constructor on the user input. The details of this
+step are further described below.
+
+![Step 2](team/aiman_assignment/step2.png)
+
+__Step 3.__ The `CommandParser` class replaces the `Assign Susan ## 301` in the user input with an empty string,
+leaving just `Susan ## 301`. Then, an instance of `AssignHousekeeperCommand` is created which extends the class `Command`.
```

Comment:
For the diagram of step2, consider changing the relationship between `Duke` and `CommandParser` from dependence to the association. I think it should be association if you are using the object of another class?